src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/CreateIndexAttemptBuilder.java

```diff
@@ -0,0 +1,90 @@
+package io.stargate.sgv2.jsonapi.service.operation.tables;
+
+import com.datastax.oss.driver.api.core.type.DataType;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableSchemaObject;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import io.stargate.sgv2.jsonapi.util.CqlIdentifierUtil;
+
+/** Builder for a {@link CreateIndexAttempt}. */
+public class CreateIndexAttemptBuilder {
+  private int position;
+  private TableSchemaObject schemaObject;
+  private String columnName;
+  private DataType dataType;
+  private String indexName;
+  private CreateIndexAttempt.TextIndexOptions textIndexOptions;
+  private CreateIndexAttempt.VectorIndexOptions vectorIndexOptions;
+  private boolean ifNotExists;
+
+  public CreateIndexAttemptBuilder(int position, TableSchemaObject schemaObject) {
```

Comment:
Is "columnName" not known when creating builder? If it is (being mandatory piece of information anyway), should just pass it along to constructor and set most of fields, make them immutable?